incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -515,6 +509,9 @@ func (r *ObjectReconciler) exclude(log logr.Logger, inst *unstructured.Unstructu
 	}
 
 	switch {
+	case r.Mode == api.Remove:
```

Comment:
Thanks for the suggestion! Please see details in my reply below. I listed the cases that HNC will not handle the objects in the `exclude` method and refer to the comment of `exclude` whenever I need, instead of restate the cases again. In this way, we can prevent changing multiple places when we want to add/remove cases in `exclude` method in future. 

> We should update the comment around line 297 that calls this function as well:
> 
> ```
> 	// Note that we only call exclude() here, not in syncPropagated, because we'll never propagate an
> 	// *uncreated* excluded object, and if an excluded object somehow got propagated, we do want to
> 	// delete it.
> ```
> 
> This comment makes it seems like it's some crazy fluke if we somehow propagated an object that should be excluded, but with this change, it becomes a far more normal case. So we should update that comment to cover the new usecase as well.

Updated along with the comment on top of `if r.exclude(log, src)`. Feel free to take a look.

> Should we also change the `exclude()` function name to be called something else? Maybe `isPropagated` (and change the sense of the boolean returned value)?

I don't think so? IMO `exclude` is very precise: we will exclude those cases from being handled by HNC. `isPropagated` might not be accurate because sometimes an object is not propagated when it is excluded (e.g., we set a type as "remove" when we create object reconciler for that type. In this case, there will not be any propagated copies). Or maybe we can change it to `shouldPropagate` then reverse the boolean return value. WDYT?

> Also, just after that comment, we call `r.syncDeletedSource`, which I think we only used to call if the source was indeed deleted. But now there can be _other_ reasons to call it, so maybe we want to call this something more like `r.syncRemovedSource`, and we can carefully comment that it can be called if the source is deleted, if the type's mode is set to "removed," or if it's excluded for any other reason.

Changed to `syncRemovedSource` and updated the comment. Right now there are two cases when we call it: the source object is deleted by users and the source object will no longer be handled by HNC (which include all the cases we talk about for `exclude`). I updated the comment to reflect that.

> Finally, maybe update the comment on line 277 as well, where we actually delete the propagated copies:
> 
> ```
> 	// Return the action to delete the obsolete copy if there's no source in the ancestors.
> ```
> 
> This also kindof implies that we're deleting the copy because the source was deleted, but it might be worth adding that this can also occur for multiple reasons (it's fine to point the user to the comments to another function).

Updated the comments

Reply:
lgtm!